x-pack/legacy/plugins/apm/public/components/shared/LocalUIFilters/index.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiTitle,
+  EuiSpacer,
+  EuiHorizontalRule,
+  EuiButtonEmpty
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import styled from 'styled-components';
+import { Filter } from './Filter';
+import { useLocalUIFilters } from '../../../hooks/useLocalUIFilters';
+import { TransactionTypeFilter } from './TransactionTypeFilter';
+
+type Props = Parameters<typeof useLocalUIFilters>[0] & {
+  showCount?: boolean;
+} & (
+    | {
+        showTransactionTypeFilter: true;
+        allowEmptyTransactionType?: boolean;
+        transactionTypes: string[];
+      }
+    | {});
```

Comment:
I think this type declaration would be better if explicitly written out instead of inferred.